lib/jms/connection.rb

```diff
@@ -178,11 +177,20 @@ def initialize(params = {})
       connection_factory = nil
       factory = options.delete(:factory)
       if factory
+        # If factory check if oracle is needed.
+        if (factory.include? 'AQjmsFactory')
```

Comment:
I agree that connecting to Oracle AQ should be done by JNDI if possible (e.g. Oracle 11/12). However I tried to connect to Oracle 8 - and in this case the implementation doesn't work as expected. At least not if I specify the factory as in the  sample - https://github.com/reidmorrison/jruby-jms/blob/master/examples/jms.yml#L128 - using ":factory: 'JMS::OracleAQConnectionFactory'". So I think it might be a good idea to also check for factory.include?('OracleAQConnectionFactory')). Thank you for your quick and detailed answer.